include/Inventor/errors/SoDebugError.h

```diff
@@ -39,8 +39,12 @@
 // Avoid problem with Microsoft Win32 API headers (yes, they actually
 // #define ERROR -- in wingdi.h).
 #if defined(ERROR)
-#define SODEBUGERROR_STORE_ERROR_DEF ERROR
```

Comment:
Rather than defining an additional enum I would use the following sequence in the `SoDebugError` header and source files:

```
#if defined(ERROR) && defined(_WIN32)
#pragma push_macro("ERROR")
#undef ERROR
#endif

< code that defines enum in header and that uses enum in source >

#if defined(_WIN32)
#pragma pop_macro("ERROR")
#endif
```

This will save the value of the `ERROR` macro on the stack with the `push_macro` command, undef the macro in the current context and restore its previous value after the `pop_macro` command.
Although `push_macro` and `pop_macro` are Microsoft preprocessor extensions (defined at least since VS6 from 1998) they are also widely supported from in Intel C++, GCC and Clang compilers.

Reply:
It looks good, but `SoDebugError::ERROR` will become an error then?

Reply:
Not in `SoDebugError.h` and `SoDebugError.cpp`, the `ERROR` preprocessor token is undefined and a new enum constant `SoDebugError::ERROR` can be defined without conflicts. But in other sources that eval `SoDebugError::ERROR` **and** include `windows.h` you need to insert the same sequence. A quick grep in the entire Coin3d repository revealed only one place in `SoWin.cpp` where `SoDebugError::ERROR` is used. This `push_macro`/`pop_macro` is used for instance in Google protobuf (see [here](https://sourcegraph.com/github.com/protocolbuffers/protobuf/-/blob/src/google/protobuf/port_def.inc) )